lib/ansible/modules/database/postgresql/postgresql_slot.py

```diff
@@ -0,0 +1,226 @@
+    C(python-psycopg2) packages on the remote host before using this module.
+requirements: [ psycopg2 ]
+author: "John Scalia (@jscalia)"
+'''
+
+EXAMPLES = '''
+# Adds physical_slot_one to the cluster running on target host default port 5432
+- postgresql_slot:
+    slot_name: physical_slot_one
+    state: present
+
+# Add a logical_slot_one to the database "acme" on target host default port 5432
+- postgresql_slot:
+    slot_name: logical_slot_one
+    type: logical
+    state: present
+    decoder: custom_decoder_one
+'''
+
+RETURN = '''
+slot_name:
+    description: Name of the slot
+    returned: success, changed
+    type: str
+    sample: "physical_slot_one"
+'''
+
+import traceback
+
+try:
+    import psycopg2
+    import psycopg2.extras
+except ImportError:
+    postgresqldb_found = False
+else:
+    postgresqldb_found = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+class NotSupportedError(Exception):
+    pass
+
+
+# ===========================================
+# PostgreSQL module specific support methods.
+#
+
+def slot_exists(cursor, slot):
+    query = "SELECT * FROM pg_replication_slots WHERE slot_name='%s'" % slot
+    cursor.execute(query, {'slot': slot})
+    return cursor.rowcount == 1
+
+
+def slot_delete(cursor, slot):
+    if slot_exists(cursor, slot):
+        query = "SELECT pg_drop_replication_slot('%s')" % slot
+        cursor.execute(query)
+        return True
+    else:
+        return False
+
+
+def slot_create_physical(cursor, slot):
+    if not slot_exists(cursor, slot):
+        query = "SELECT pg_create_physical_replication_slot('%s')" % slot
+        cursor.execute(query)
+        return True
+    else:
+        return False
+
+
+def slot_create_logical(cursor, slot, decoder):
+    if not slot_exists(cursor, slot):
+        query = "SELECT pg_create_logical_replication_slot('%s', '%s')" % (slot, decoder)
+        cursor.execute(query)
+        return True
+    else:
+        return False
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(default="postgres"),
+            login_password=dict(default="", no_log=True),
+            login_host=dict(default="localhost"),
+            port=dict(default="5432"),
+            db=dict(required=False),
+            slot_name=dict(required=True),
+            type=dict(default="physical", choices=["physical", "logical"]),
+            decoder=dict(default="test_decoding"),
+            state=dict(default="present", choices=["absent", "present"]),
+        ),
+        supports_check_mode=True
+    )
+
+    if not postgresqldb_found:
+        module.fail_json(msg="the python psycopg2 module is required")
+
+    db = module.params["db"]
+    slot = module.params["slot_name"]
+    type = module.params["type"]
```

Comment:
It's generally not a good idea to mask [built-in](https://docs.python.org/2/library/functions.html#type) functions.
One more reason to pick `slot_type` as per the [docs](https://www.postgresql.org/docs/current/view-pg-replication-slots.html).